### DISCUSSION TITLE
OPA Input Output Hierarchy

### DISCUSSION THREAD

**Discussion by Ronnie-personal** [open-policy-agent/community#379] [OPA and Rego] [answered]
Could you confirm if OPA can only accept one single input json file?

I think OPA can accept multiple data json files, but there is no module or namespace for the data, it's flat structure and is all under 'data.'
Is my understanding correct?

Thanks!

**Comment by anderseknert** [comment, marked as answer]
Yes, your understanding is correct. The data is "namespaced" by the attribute names alone, and all data is merged into a single document once loaded into OPA.

**Reply by Ronnie-personal** [reply]
Thanks for the prompt reply.

I am thinking of using dedicated rego files to hold JSON data instead of using JSON files.

What do you think of the approach?

Rego file supports module, so that the json data is namespaced under package name.

**Reply by anderseknert** [reply]
Sure! Use whatever works best for you 😃
